### ISSUE TITLE
Is it possible to control the visibility of a group?

### ISSUE THREAD

**Issue by Jack-Badger** [codestack-net-dev/vpages-sw#20]
My Macro Feature sometimes needs only 2 of it's 3 dimensions.

For example with these options checked I would like to hide the D3 group.

![image](https://user-images.githubusercontent.com/46345306/68483434-7784c900-0233-11ea-9e92-bebb19d2074c.png)

Is this something that can be done, maybe by getting the  IPropertyManagerPageGroup or is it only possible to work with Controls?

Best Regards
Rob

**Comment by Jack-Badger** [comment]
I managed to get it working this way..

![image](https://user-images.githubusercontent.com/46345306/68500400-8e3b1800-0253-11ea-8f7c-08fe1f312368.png)

Do you think that's ok?

**Comment by codestackdev** [comment]
Hi Rob,

Good workaround. But there is built-in mechanism to handle [dependencies](https://www.codestack.net/labs/solidworks/swex/pmpage/controls/dependencies/). And your code can look similar to the one below:

~~~cs
public class PmpData
{
    public class Group
    {
        public string Text { get; set; }
        public double Number { get; set; }
    }

    [DependentOn(typeof(GroupBoxVisibilityDepHandler), nameof(CreateSolidBox))]
    public Group D1 { get; set; }

    [DependentOn(typeof(GroupBoxVisibilityDepHandler), nameof(CreateSurfaceBox))]
    public Group D2 { get; set; }

    [DependentOn(typeof(GroupBoxVisibilityDepHandler), nameof(CreatePlanarBox))]
    public Group D3 { get; set; }

    [ControlTag(nameof(CreateSolidBox))]
    public bool CreateSolidBox { get; set; }

    [ControlTag(nameof(CreateSurfaceBox))]
    public bool CreateSurfaceBox { get; set; }

    [ControlTag(nameof(CreatePlanarBox))]
    public bool CreatePlanarBox { get; set; }
}

public class GroupBoxVisibilityDepHandler : DependencyHandler
{
    protected override void UpdateControlState(IPropertyManagerPageElementEx control, IPropertyManagerPageElementEx[] parents)
    {
        control.Visible = parents.First().GetValue<bool>();
    }
}
~~~
Please note there was a [bug](https://github.com/codestackdev/swex-pmpage/issues/21) which is now fixed in 0.7.2. Please update your nuget package.

I like the above approach as it allows to separate UI logic and do not handle it from business logic (i.e. data). Please let me know if any questions or issues.
